src/gui/generalsettings.cpp

```diff
@@ -68,6 +68,7 @@ GeneralSettings::GeneralSettings(QWidget *parent)
     connect(_ui->newFolderLimitCheckBox, &QAbstractButton::toggled, this, &GeneralSettings::saveMiscSettings);
     connect(_ui->newFolderLimitSpinBox, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), this, &GeneralSettings::saveMiscSettings);
     connect(_ui->newExternalStorage, &QAbstractButton::toggled, this, &GeneralSettings::saveMiscSettings);
+    connect(_ui->moveToTrashCheckBox, &QAbstractButton::toggled, this, &GeneralSettings::saveMiscSettings);
```

Comment:
also, the chekbox should be hidden on non-unix is the move to trash function is not implemented

Reply:
Isn't [`Q_OS_UNIX`](https://doc-snapshots.qt.io/qt5-dev/qtglobal.html#Q_OS_UNIX) defined for macOS as well? I mean macOS is also unix-based.